--- ./cnf/gac.in.orig	2013-12-01 03:17:13.000000000 -0700
+++ ./cnf/gac.in	2014-01-13 22:00:00.000000000 -0700
@@ -55,16 +55,16 @@
 ##  option is appended at the end of the link command after the .o
 ##  files to link.
 ##
-gap_bin=@gapbin@
+gap_bin=/usr/include/gap
 
 if [ "X${gap_bin}" = "X$0" ];  then
     gap_dir="../../";
     gap_compiler="./gap";
     gap_binary="./";
 else
-    gap_dir="${gap_bin}/../..";
-    gap_compiler="${gap_bin}/gap";
-    gap_binary="${gap_bin}";
+    gap_dir=/usr/lib/gap;
+    gap_compiler=/usr/bin/gap;
+    gap_binary=${gap_dir}/bin/@gaparch@;
 fi
 gap_options=""
 gap_include="${gap_dir}/src"
@@ -171,9 +171,9 @@ gap_compile () {
 ##
 c_compile () {
     echo ${c_compiler}  $3 -o $1 -I${gap_dir} \
-         -I${gap_bin} -DCONFIG_H -c $2
+         -I${gap_bin} -I${gap_binary} -DCONFIG_H -c $2
     ${c_compiler} $3 -o $1 -I${gap_dir} \
-         -I${gap_bin} -DCONFIG_H -c $2 || exit 1
+         -I${gap_bin} -I${gap_binary} -DCONFIG_H -c $2 || exit 1
 }
 
 
@@ -439,17 +439,17 @@ make_compstat () {
     objects="${gactmp}/$$compstat.o ${objects}"
 }
 
-make_tmpdir () {
-    basetmp=${TMPDIR:-/tmp}; #honor the TMPDIR environment variable.
-    gactmp="$basetmp/gac$$";
-    mkdir "$gactmp" || exit 1;
-}
+#make_tmpdir () {
+#    basetmp=${TMPDIR:-/tmp}; #honor the TMPDIR environment variable.
+#    gactmp="$basetmp/gac$$";
+#    mkdir "$gactmp" || exit 1;
+#}
 
 #On systems with mktemp(1) this function is better:
-#make_tmpdir ()
-#{
-#    gactmp=`mktemp -d -t "gacXXXXXXX"`;
-#}
+make_tmpdir ()
+{
+    gactmp=`mktemp -d -t "gacXXXXXXX"`;
+}
 
 #############################################################################
 ##
--- ./configure.orig	2013-12-01 03:17:13.000000000 -0700
+++ ./configure	2014-01-13 22:00:00.000000000 -0700
@@ -4094,7 +4094,7 @@ $as_echo "no" >&6; }
 fi
 
 
-gapdir=`pwd`
+gapdir=/usr/lib/gap
 
 
 
