--- ./lib/helpdef.gi.orig	2013-12-01 03:17:14.000000000 -0700
+++ ./lib/helpdef.gi	2014-01-13 22:00:00.000000000 -0700
@@ -329,6 +329,11 @@ HELP_BOOK_HANDLER.default.ReadSix := fun
       res.pdffile := ShallowCopy(fname);
       Add(res.formats, "pdf");
     fi;
+    fname{[Length(fname)-5..Length(fname)]} := "dvi.gz";
+    if IsExistingFile( fname ) = true then
+      res.dvifile := ShallowCopy(fname);
+      Add(res.formats, "dvi");
+    fi;
   fi;
   res.directories := Directory(fname{[1..Length(fname)-10]});  
   return res;
--- ./lib/helpview.gi.orig	2013-12-01 03:17:14.000000000 -0700
+++ ./lib/helpview.gi	2014-01-13 22:00:00.000000000 -0700
@@ -430,6 +430,27 @@ show := function(file)
 end
 );
 
+# View HTML, DVI, or PDF files with xdg-open, which opens documents with the
+# user's preferred application.
+HELP_VIEWER_INFO.xdghtml := rec(
+type := "url",
+show := function(url)
+  Exec(Concatenation("xdg-open \"", url, "\""));
+end
+);
+HELP_VIEWER_INFO.xdgdvi := rec(
+type := "dvi",
+show := function(file)
+  Exec(Concatenation("xdg-open \"", file, "\""));
+end
+);
+HELP_VIEWER_INFO.xdgpdf := rec(
+type := "pdf",
+show := function(file)
+  Exec(Concatenation("xdg-open \"", file, "\""));
+end
+);
+
 #############################################################################
 ##
 #F  SetHelpViewer(<viewer>):  Set the viewer used for help
